src/qutip_qip/qir.py

```diff
@@ -0,0 +1,180 @@
+    # branches.
+    def append_operation(module: pqg.SimpleModule, builder: pqg.BasicQisBuilder, op: Gate):
+        if op.classical_controls:
+            result = op.classical_controls[0]
+            value = "zero" if op.classical_control_value == 0 else "one"
+            # Pull off the first control and recurse.
+            op_with_less_controls = Gate(**op.__dict__)
+            op_with_less_controls.classical_controls = op_with_less_controls.classical_controls[1:]
+            branch_body = {value: (lambda: append_operation(module, builder, op_with_less_controls))}
+            builder.if_result(module.results[result], **branch_body)
+            return
+
+        if op.controls:
+            if op.name not in ("CNOT", "CZ") or len(op.controls) != 1:
+                raise NotImplementedError(
+                    "Arbitrary controlled quantum operations are not yet supported."
+                )
+
+        if op.name == "X":
+            builder.x(module.qubits[op.targets[0]])
+        elif op.name == "Y":
+            builder.y(module.qubits[op.targets[0]])
+        elif op.name == "Z":
+            builder.z(module.qubits[op.targets[0]])
+        elif op.name == "S":
+            builder.s(module.qubits[op.targets[0]])
+        elif op.name == "T":
+            builder.t(module.qubits[op.targets[0]])
+        elif op.name == "SNOT":
+            builder.h(module.qubits[op.targets[0]])
+        elif op.name == "CNOT":
+            builder.cx(module.qubits[op.controls[0]], module.qubits[op.targets[0]])
+        elif op.name == "RX":
+            builder.rx(op.control_value, module.qubits[op.targets[0]])
+        elif op.name == "RY":
+            builder.ry(op.control_value, module.qubits[op.targets[0]])
+        elif op.name == "RZ":
+            builder.rz(op.control_value, module.qubits[op.targets[0]])
+        elif op.name in ("CRZ", "TOFFOLI"):
+            raise NotImplementedError(
+                "Decomposition of CRZ and Toffoli gates into base " +
+                "profile instructions is not yet implemented."
+            )
+        else:
+            raise ValueError(
+                f"Gate {op.name} not supported by the basic QIR builder, " +
+                "and may require a custom declaration."
+            )
+    fmt = QirFormat.ensure(format)
+
+    module = pqg.SimpleModule(module_name, circuit.N, circuit.num_cbits or 0)
+    builder = pqg.BasicQisBuilder(module.builder)
+
+    for op in circuit.gates:
+        # If we have a QuTiP gate, then we need to convert it into one of
+        # the reserved operation names in the QIR base profile's quantum
+        # instruction set (QIS).
+        if isinstance(op, Gate):
+            # TODO: Validate indices.
+            append_operation(module, builder, op)
+
+        elif isinstance(op, Measurement):
+            # TODO: Validate indices.
+            if op.name == "Z":
+                builder.m(module.qubits[op.targets[0]], module.results[op.classical_store])
```

Comment:
The measurement in qutip always measure in the Z basis. Regardless of the `name`. So I guess this is not necessary.

Reply:
Ah, got it; removed that condition, then. Thanks!